src/Catalyst.Core.Lib/IO/Handlers/FlushPipelineHandler.cs

```diff
@@ -0,0 +1,42 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Reflection;
+using System.Threading.Tasks;
+using DotNetty.Transport.Channels;
+using Serilog;
+
+namespace Catalyst.Core.Lib.IO.Handlers
+{
+    public sealed class FlushPipelineHandler<T> : OutboundChannelHandlerBase<T>
+    {
+        // ReSharper disable once StaticMemberInGenericType
+        private static readonly ILogger Logger = Log.Logger.ForContext(MethodBase.GetCurrentMethod().DeclaringType);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A static field in a generic type is not shared among instances of different close constructed types.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)